Parkinson/Startup/APHAppDelegate.m

```diff
@@ -221,16 +328,393 @@ - (NSArray *)allSetTextBlocks
+                                                                        @"horizontalAccuracyUnit",
+                                                                        @"verticalAccuracy",
+                                                                        @"verticalAccuracyUnit"];
+    APCPassiveDisplacementTrackingDataUploader* displacementSinker  = [[APCPassiveDisplacementTrackingDataUploader alloc]
+                                                                       initWithIdentifier:@"locationTracker"
+                                                                       columnNames:locationColumns
+                                                                       operationQueueName:@"APCDisplacement Tracker Sink"
+                                                                       andDataProcessor:nil];
+    [locationCollector setReceiver:displacementSinker];
+    [locationCollector setDelegate:displacementSinker];
+    [locationCollector start];
+    [self.passiveDataCollector addDataSink:locationCollector];
+}
+
+- (void)configureObserverQueries
+{
+    NSDate* (^LaunchDate)() = ^
     {
-        [self.passiveDataCollector addTracker: locationTracker];
+        APCUser*    user        = ((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.currentUser;
+        NSDate*     consentDate = nil;
+        
+        if (user.consentSignatureDate)
+        {
+            consentDate = user.consentSignatureDate;
+        }
+        else
+        {
+            NSFileManager*  fileManager = [NSFileManager defaultManager];
+            NSString*       filePath    = [[self applicationDocumentsDirectory] stringByAppendingPathComponent:@"db.sqlite"];
+            
+            if ([fileManager fileExistsAtPath:filePath])
+            {
+                NSError*        error       = nil;
+                NSDictionary*   attributes  = [fileManager attributesOfItemAtPath:filePath error:&error];
+                
+                if (error)
+                {
+                    APCLogError2(error);
+                    
+                    consentDate = [[NSDate date] startOfDay];
+                }
+                else
+                {
+                    consentDate = [attributes fileCreationDate];
+                }
+            }
+        }
+        
+        return consentDate;
+    };
+    
+    NSString*(^QuantityDataSerializer)(id) = ^NSString*(id dataSample)
+    {
+        HKQuantitySample*   qtySample           = (HKQuantitySample *)dataSample;
+        NSString*           startDateTimeStamp  = [qtySample.startDate toStringInISO8601Format];
+        NSString*           endDateTimeStamp    = [qtySample.endDate toStringInISO8601Format];
+        NSString*           healthKitType       = qtySample.quantityType.identifier;
+        NSString*           quantityValueRep    = [NSString stringWithFormat:@"%@", qtySample.quantity];
```

Comment:
Same comment applies here as in other projects. Should not be doing this to recover the canonical unit. Instead, define what unit you want to use for each type and use doubleValueForUnit: on the quantity.